mne/viz/epochs.py

```diff
@@ -65,71 +67,252 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+        case "Reds" is used.
     fig : matplotlib figure | None
         Figure instance to draw the image to. Figure must contain two axes for
         drawing the single trials and evoked responses. If None a new figure is
         created. Defaults to None.
-    axes : list of matplotlib axes | None
+    axes : list of matplotlib axes | dict of lists of matplotlib Axes | None
         List of axes instances to draw the image, erp and colorbar to.
         Must be of length three if colorbar is True (with the last list element
         being the colorbar axes) or two if colorbar is False. If both fig and
-        axes are passed an error is raised. Defaults to None.
+        axes are passed an error is raised.
+        If `groupby` is a dict, this can also be a dict of lists of axes,
+        with the keys matching. In that case, the provided axes will be used
+        for the corresponding groups.
+        Defaults to None.
     overlay_times : array-like, shape (n_epochs,) | None
         If not None the parameter is interpreted as time instants in seconds
         and is added to the image. It is typically useful to display reaction
         times. Note that it is defined with respect to the order
         of epochs such that overlay_times[0] corresponds to epochs[0].
+    combine : None | str | callable
+        If None, return one figure per pick. If not None, aggregate over
+        channels via the indicated method. If str, must be one of "mean",
+        "std" and "gfp", in which case the average, the standard deviation or
+        the GFP over channels are calculated and plotted.
+        If callable, it must accept one positional input, the data
+        in the format (n_epochs, n_channels, n_times). It must return an
+        array (n_epochs, n_times). For example:
+
+            `combine = lambda data: np.median(data, 1)`
+
+    groupby : None | str | dict
+        If not None, combine must not be None. In this case, combining happens
+        over channel groups defined by this parameter.
+        If str, must be "type", in which case one figure per channel type is
+        returned (combining within channel types).
+        If a dict, the values must be picks and one figure will be returned
+        for each entry, aggregating over the corresponding pick groups. This
+        is useful for e.g. ROIs. Each entry must contain only one channel type.
+        For example:
+
+           `groupby = dict(Left_ROI=[1, 2, 3, 4], Right_ROI=[5, 6, 7, 8])`
 
     Returns
     -------
     figs : lists of matplotlib figures
         One figure per channel displayed.
     """
-    from scipy import ndimage
     units = _handle_default('units', units)
     scalings = _handle_default('scalings', scalings)
 
-    import matplotlib.pyplot as plt
+    if groupby is not None and combine is None:
+        raise ValueError("If groupby is not None, combine must not be None.")
```

Comment:
Not sure. When we came up with this, "mean" wasn't even in the discussion, it was about GFPs.

I like defaults that make good plots with few input, but I'd def. wait for input from e.g. @agramfort, who proposed `groupby`, first.